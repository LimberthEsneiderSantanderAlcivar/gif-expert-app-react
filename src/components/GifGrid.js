import React from 'react'

const GifGrid = ({category}) => {
    
    const getGifs = async()=>{
        const url = 'https://api.giphy.com/v1/gifs/search?q=rick and morty&limit=10&api_key=xbcFPFzqBmawj0eB4EBMmlBds5PZMETW'
        const resp = await fetch(url);
        const {data} = await resp.json();

        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        });

        console.log(gifs);
    }

    getGifs()
    return (
        <>
            <h3>{category}</h3>   
        </>
    )
}

export default GifGrid
