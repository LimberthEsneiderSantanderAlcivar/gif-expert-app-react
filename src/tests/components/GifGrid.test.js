import {shallow} from 'enzyme'
import GifGrid from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('preubas del componente GifGrid', () => {
    const category = 'Shingeky No Kyojin';


    test('debe hacer match con el snapshot', () => {
        useFetchGifs.mockReturnValue({
            data:[],
            loading:false
        })
        const wrapper = shallow(<GifGrid category={category}/>)
        expect(wrapper).toMatchSnapshot();
    });
    
    
    test('debe de mostrar items cuando se cargan las imagenes useFetchGifs', () => {
        const gifs =[{
            id:'abc',
            url: 'https://asdas.com',
            title:'cualquier cosa'
        }]
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false
        })
        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);                
    })
    
})
