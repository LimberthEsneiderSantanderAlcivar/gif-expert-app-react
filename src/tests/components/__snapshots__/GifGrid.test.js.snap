// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preubas del componente GifGrid debe de mostrar items cuando se cargan las imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Shingeky No Kyojin
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="cualquier cosa"
      url="https://asdas.com"
    />
  </div>
</Fragment>
`;

exports[`preubas del componente GifGrid debe hacer match con el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Shingeky No Kyojin
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
